demo/rust/micdemo/src/main.rs

```diff
@@ -0,0 +1,299 @@
+/*
+    Copyright 2018-2021 Picovoice Inc.
+
+    You may not use this file except in compliance with the license. A copy of the license is located in the "LICENSE"
+    file accompanying this source.
+
+    Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+    an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+    specific language governing permissions and limitations under the License.
+*/
+
+use clap::{App, Arg, ArgGroup};
+use ctrlc;
+use miniaudio;
+use pv_porcupine::{pv_keyword_paths, pv_library_path, pv_model_path, Porcupine, KEYWORDS};
+use std::path::PathBuf;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::time::SystemTime;
+
+static LISTENING: AtomicBool = AtomicBool::new(true);
+
+fn process_audio_chunk(
+    samples: &[i16],
+    porcupine: &mut Porcupine,
+    buffer: &mut Vec<i16>,
+    keywords: &[String],
+) {
+    buffer.extend_from_slice(samples);
+
+    while buffer.len() >= porcupine.frame_length() as usize && LISTENING.load(Ordering::SeqCst) {
+        let frame: Vec<i16> = buffer.drain(..porcupine.frame_length() as usize).collect();
+        let result = porcupine.process(&frame).unwrap();
+        if result >= 0 {
+            println!(
+                "[{}] Detected {}",
+                SystemTime::now()
+                    .duration_since(SystemTime::UNIX_EPOCH)
```

Comment:
Can you make the time more readable?

Reply:
With a package dependency, sure :)

Reply:
dang, really? Might be worth it for the demo. The timestamp just looks pretty gnarly. Either that or don't show the timestamp.